resources/windows/wix/service-filter.xslt.erb

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0"
+ xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+ xmlns:wix="http://schemas.microsoft.com/wix/2006/wi">
+  <!-- https://ahordijk.wordpress.com/2013/03/26/automatically-add-references-and-content-to-the-wix-installer/ -->
+  <!-- http://www.chriskonces.com/using-xslt-with-heat-exe-wix-to-create-windows-service-installs/ -->
+  <xsl:output method="xml" indent="yes" />
+  <!--<xsl:strip-space elements="*"/>-->
+  <xsl:template match="@*|node()">
+    <xsl:copy>
+      <xsl:apply-templates select="@*|node()"/>
+    </xsl:copy>
+  </xsl:template>
+  <!-- Filter out all Service File Executables from the Harvest (heat) run as these are specified in the transformed service.components.wxs file -->
+  <%- get_services.each do |service| -%>
+      <xsl:template match="wix:Component[wix:File[@Source='<%= service.options.first[:file][:source] %>']]" />

```

Comment:
If we have multiple entries of one source
```
stagedir\sys\ruby\bin\ruby.exe
stagedir\sys\ruby\bin\ruby.exe
stagedir\sys\ruby\bin\ruby.exe
stagedir\sys\ruby\bin\ruby.exe
stagedir\sys\ruby\bin\ruby.exe
```
we want to specify it only once.

This may happen if we have multiple services using the same source. I.e., the puppet services all use `stagedir\sys\ruby\bin\ruby.exe`. They all have unique identifiers, but if we are passing that path to this file, then we will have multiple entries of stagedir\sys\ruby\bin\ruby.exe. Or if we have different components that have services that use the same source. That isn't currently the case, but if mco used `stagedir\sys\ruby\bin\ruby.exe` rather than `stagedir\sys\ruby\bin\rubyw.exe` we would have duplicate sources.

Ruby has the ability to remove any duplicates from an array (http://docs.ruby-lang.org/en/2.1.0/Array.html#method-i-uniq). The method is called `uniq`.

Have you tested this with https://github.com/puppetlabs/puppet-agent/pull/519 ? If you have multiple service entries for one component, you're going to miss the additional ones. We have no guarentee that multiple services in a component will use the same source. In this current example, you're only taking the first service entry and ignoring the rest.